prov/sockets/src/sock_ep.c

```diff
@@ -1394,6 +1394,8 @@ struct fi_info *sock_fi_info(enum fi_ep_type ep_type, struct fi_info *hints,
 	if (src_addr) {
 		memcpy(info->src_addr, src_addr, sizeof(struct sockaddr_in));
 		info->src_addrlen = sizeof(struct sockaddr_in);
+	} else {
+		sock_get_src_addr_from_hostname(src_addr, NULL);
 	}
```

Comment:
Issue here - In sock_get_src_addr_from_hostname(), you are doing a memcpy to src_addr (NULL in this case), which will result in a seg fault. 

The issue that coverity reported is because you are passing src_addr (which can be NULL) to sock_set_fabric_attr, where it is dereferenced later.